### DISCUSSION TITLE
YAxis zoom

### DISCUSSION THREAD

**Discussion by permiakover** [patrykandpatrick/vico#1058] [Ideas]
Hi! How could YAxis scroll be enabled?

**Comment by ghost** [comment]
Hello! Unfortunately, Vico doesn’t include such functionality. We don’t currently plan on adding it, but we may consider doing so in the future. We understand that vertical scrolling would be required for full-screen financial charts, for example, but Vico is intended for more conventional use cases.

**Comment by matej-marjanovic** [comment]
Please add to the documentation that zoom and scroll are only available for X-axis.
As it is, it took me some time to confirm that Y-axis zoom and scrolling is not available.

Maybe you could reference a link for people to upvote if they are interested in this feature.

**Comment by patrickmichalik** [comment]
We’ll look into updating the documentation accordingly, @matej-marjanovic. Regarding the feature request, I’m afraid that, as per the comment above, we don’t plan on implementing vertical scrolling as of now.

**Comment by Shusek** [comment]
In theory, for the version using Compose, dynamically providing instances of the "RangeProvider" should work. Of course, due to recompositions, it won't be particularly smooth, it's currently feasible as long as linking the zoom to both the x and y axes simultaneously isn't an issue.

Eg.
```
val rangeProvider = remember(zoomValue){
CartesianLayerRangeProvider{
// calculation with zoom value
}
}
```

**Comment by Gowsky** [comment]
Hello, @Shusek.

> dynamically providing instances of the "RangeProvider" should work

As discussed in the [guide](https://www.patrykandpatrick.com/vico/guide/stable/core/cartesian-charts/cartesian-layer#cartesianlayerrangeprovider), such usage is incorrect and wouldn't produce the expected result. Instead, you'd need to have a single `CartesianLayerRangeProvider` that reads the _y_ range from extras, which you'd update to control the _y_-axis zoom. This would be functional but not perfectly responsive, since the zoom would be handled asynchronously at the data level instead of in real time at the UI level.

> due to recompositions, it won't be particularly smooth

Recompositions are normal and aren't a concern. Any change to what's displayed involves a recomposition. For example, when you interact with a `Slider`, it recomposes as the value changes.

> as long as linking the zoom to both the x and y axes simultaneously isn't an issue

You don't necessarily have to do this. You could add a gesture modifier to the `CartesianChartHost` and detect vertical pinches for independent _x_-axis and _y_-axis zoom. The only problem is the potential for slightly lower responsiveness, as mentioned above.

**Comment by patrickmichalik** [comment]
As we’re shifting to a more community-driven approach, I’ll be reopening this as a discussion in the “Ideas” category. Please upvote the discussion if you’d like to see vertical scroll and zoom added.

**Comment by yuval-ashcer** [comment]
Any updates on this one?
If not, any possible workarounds?
